### DISCUSSION TITLE
I get a "malformed array literal" error when I pass an enum value

### DISCUSSION THREAD

**Discussion by mkkeems** [mikro-orm/mikro-orm#4128] [General]
Hello,

I am using Mikro-orm 5.6 with postgresql and typescript and need help with an error i don't quite understand.

I have an entity `Project`, with property `unit`, which is an array of enum values:
```
 @Enum({ items: () => Unit, array: true, nullable: true })
  unit: Unit[];
```
while `Unit` is an enum:
```
export enum Unit {
  MKT = "MKT",
  UX = "UX",
  PROD = "PROD",
  CX = "CX",
}
```

When I try to find all projects where `unit` is ['MKT'] like this:
```
    const projects = await em.find(Project, {
      unit: [Unit.MKT],
        });
```
there is no typescript error in the code above 

BUT
I get this error when the query is run:
`"select \"p0\".* from \"project\" as \"p0\" where \"p0\".\"unit\" in ('MKT') - malformed array literal: \"MKT\""`

There is no problem in creating projects doing something like `em.create(Project, { unit: [Unit.MKT, Unit, UX]..})`
However, when i try to query projects by the unit value, i get this error.
(same error occurs when i do: `const projects = await em.find(Project, {unit: ["MKT"]});`)

Is there something wrong with how I defined the unit property on Project entity?

In case this information is helpful, this is from the migration file that involves the "unit" column of "Project" table:
```
    this.addSql(
      'alter table "project" alter column "unit" type text[] using ("unit"::text[]);'
    );
```

and the schema snapshot looks like this:
```
        "unit": {
          "name": "unit",
          "type": "text[]",
          "unsigned": false,
          "autoincrement": false,
          "primary": false,
          "nullable": true,
          "enumItems": [
            "MKT",
            "UX",
            "PROD",
            "CX"
          ],
          "mappedType": "array"
        },
```

**Comment by B4nan** [comment]
Sounds like a bug, but I have one question - what do you expect from your query? I believe you should be using one of the postgres array operators here (which on their own probably wont help), as even if the array literal would be correct, your query would fail on inability of postgres to compare array with a single value - as the right-hand side of your query will get translated to `where in` query automatically.

<img width="463" alt="image" src="https://user-images.githubusercontent.com/615580/224949685-1591046a-84fd-44c0-b666-170fa5dcc7ec.png">

What could help is if you specify the `EnumArrayType` explicitly, as this sounds like the information is lost during discovery or metadata caching:

```ts
@Enum({ items: () => Unit, array: true, nullable: true, type: EnumArrayType })
unit: Unit[];
```

**Reply by mkkeems** [reply]
Thank you for your very quick response! 😄
My intention was to query all projects where value of unit had an exact match of [Unit.MKT].

I just got around this error by doing this:
`const projects = await em.find(Project, { unit: { $eq: [Unit.MKT] } });`

Thank you so much!

On a side note, should adding `type: EnumArrayType` as you suggested be detectable by the `npx mikro-orm migration:create` cli command?
Or is this just something additionally defined as reference for mikro-orm?

**Reply by B4nan** [reply]
It shouldn't be needed, I guess you are using ts-morph and metadata caching, right? That's what breaks it I think, please share more about your setup, ORM config should be enough.
